src/main/java/com/three_tech_solutions/slot_app/mappers/StudentMapper.java

```diff
@@ -0,0 +1,42 @@
+package com.three_tech_solutions.slot_app.mappers;
+
+import com.three_tech_solutions.slot_app.data.models.Plan;
+import com.three_tech_solutions.slot_app.data.models.User;
+import com.three_tech_solutions.slot_app.dto.StudentRequestDTO;
+import com.three_tech_solutions.slot_app.data.models.Student;
+import com.three_tech_solutions.slot_app.dto.StudentResponseDTO;
+import org.springframework.stereotype.Service;
+
+@Service
+public class StudentMapper {
+    /**
+     * Mapea un StudentRequestDTO a la entidad Student.
+     * El Plan y los Payments se reciben como parámetros, ya deben ser buscados o generados en el Service.
+     */
+
+    public Student toStudent(StudentRequestDTO studentDTO, Plan plan, User user) {
+        Student student = new Student();
+
+        student.setName(studentDTO.getName());
+        student.setLastname(studentDTO.getLastName());
+        student.setPhoneNumber(studentDTO.getCellphoneNumber());
+        student.setBirthday(studentDTO.getBirthday());
+        student.setAdmissionDate(studentDTO.getAdmissionDate());
+        student.setPathologies(studentDTO.getPathologies());
+        student.setEnabled(true);// Activo por defecto
+        student.setPlan(plan);
+        student.setUser(user);
+
+        return student;
```

Comment:
Acá estas haciendo un mapeo por setter, también podrías hacer un mappeo por constructor de la siguiente manera

```suggestion
        return new Student(
                studentDTO.getName(),
                studentDTO.getLastName(),
                studentDTO.getCellphoneNumber(),
                studentDTO.getBirthday(),
                studentDTO.getPathologies(),
                studentDTO.getAdmissionDate(),
                true,
                plan,
                user
        );
```

Lo unico que tendrias que crear el constructor especifico

por ejemplo

```
    public Student(String name, String lastname, String phoneNumber, LocalDate birthday, String pathologies, LocalDate admissionDate, boolean enabled, Plan plan, User user) {
        this.name = name;
        this.lastname = lastname;
        this.phoneNumber = phoneNumber;
        this.birthday = birthday;
        this.pathologies = pathologies;
        this.admissionDate = admissionDate;
        this.enabled = enabled;
        this.plan = plan;
        this.user = user;
    }
```